### DISCUSSION TITLE
PDF output maintaining a specific PDF version.

### DISCUSSION THREAD

**Discussion by optimisticwalker** [ImageMagick/ImageMagick#7604] [Help]
### ImageMagick version

7

### Operating system, version and so on

Linux

### Description

Hi,

There isn't enough resource available about how to output a PDF file maintaining a specific PDF version.

Can anyone give me a clue about to maintain a specific PDF version when outputting a PDF file?

Example:  PDF version 1.3 & PDF version 1.4, etc.

Thanks,

**Comment by fmw42** [comment]
You question is not clear.  What was your command?  Was the input PDF version not maintained in the output PDF?

**Reply by optimisticwalker** [reply]
Unfortunately, no. Conversion always outputs a higher version when units, density and background are specified.

Example: convert -units PixelsPerInch -density 150 a.pdf -background yellow b.pdf

I don't find any remarkable documentation about it.

Thanks.

**Comment by snibgo** [comment]
IM's rules about the version number of output PDFs seems complicated. See https://github.com/ImageMagick/ImageMagick/blob/main/coders/pdf.c

**Reply by optimisticwalker** [reply]
Thanks for the reference @snibgo.
